lib/client/client.js

```diff
@@ -792,6 +797,8 @@ Client.prototype.connect = function connect () {
 
   // Establish basic socket connection
   function connectSocket (cb) {
+    self.nextServer = (self.nextServer + 1) % self.servers.length
+    self.url = self.servers[self.nextServer]
```

Comment:
This seems like a bad idea to use a mutating parameter when dealing with async code.

What stops this from changing before one of the async functions uses it?

Reply:
Both (`self.nextServer` & `self.url`) are updated only at the start of `connectSocket` function. Reading both is done in synchronous way (https://github.com/wision/node-ldapjs/blob/multi-servers/lib/client/client.js#L838-L839 and https://github.com/wision/node-ldapjs/blob/multi-servers/lib/client/client.js#L865, where `initSocket` is called synchronously in `connectSocket` https://github.com/wision/node-ldapjs/blob/multi-servers/lib/client/client.js#L848)

Reply:
It looks like `self.url` is only used in [`initSocket`](https://github.com/wision/node-ldapjs/blob/multi-servers/lib/client/client.js#L863) there is no reason for it to be global especially since it is not necessarily the url that will be used to connect.

If someone calls `client.connect()` then `client.url` could be the wrong url.

`self.nextServer` looks like it does have to be global but it doesn't have to be public.